src/contributors/MapContributor.ts

```diff
@@ -1024,12 +1027,31 @@ export class MapContributor extends Contributor {
                 const start = Date.now();
                 const cancelSubjects = this.cancelSubjects.get(searchId);
                 const lastCall = this.lastCalls.get(searchId);
+                const renderRetries = [];
                 this.resolveTiledSearchSources(newVisitedTiles, searchId, searchSource.search)
                 .pipe(
                     takeUntil(cancelSubjects && cancelSubjects.get(lastCall) ? cancelSubjects.get(lastCall) : of()),
 
                     map(f => this.computeFeatureData(f, searchSource.sources)),
                     tap(() => count++),
+                    tap(() => {
+                        const progression = count / totalcount * 100;
+                        const consumption = Date.now() - start;
+                        if (consumption > 2000) {
+                            if (progression > 25 && renderRetries.length === 0) {
```

Comment:
if progression == 80 && renderRetries.length === 0 then it will enter in the three `if`, is this expected? Or should we use `else`?

Reply:
Fixed (we have to use else)